R/utils.R

```diff
@@ -30,3 +30,18 @@ check_requires <- function(what, requires) {
 
   invisible(TRUE)
 }
+
+#' Check if we are in a knitr notebook
+#'
+#' @return `TRUE` if we are in a knitr notebook, `FALSE` otherwise
+#'
+#' @noRd
+is_knitr_notebook <- function() {
+  # if knitr is not available, assume that we are not in a notebook
+  if (!requireNamespace("knitr", quietly = TRUE)) {
```

Comment:
In my PR I updated `check_requires()` to return `TRUE`/`FALSE` rather than raise an error when you set an argument. You could use that when things are merged but this is also fine.